src/psyclone/transformations.py

```diff
@@ -563,98 +562,232 @@ def apply(self, node, options=None):
         return rval
 
 
-class OMPLoopTrans(ParallelLoopTrans):
+class OMPTargetTrans(RegionTrans):
+    '''
+    Adds an OpenMP target directive to a region of code.
+
+    For example:
+
+    >>> from psyclone.psyir.frontend.fortran import FortranReader
+    >>> from psyclone.psyir.nodes import Loop
+    >>> from psyclone.transformations import OMPTargetTrans
+    >>>
+    >>> tree = FortranReader().psyir_from_source("""
+    >>>     subroutine my_subroutine()
+    >>>         integer, dimension(10, 10) :: A
+    >>>         integer :: i
+    >>>         integer :: j
+    >>>         do i = 1, 10
+    >>>             do j = 1, 10
+    >>>                 A(i, j) = 0
+    >>>             end do
+    >>>         end do
+    >>>     end subroutine
+    >>>     """
+    >>> omptargettrans = OMPTargetTrans()
+    >>> omptargettrans.apply(tree.walk(Loop))
+
+    will generate:
+
+    .. code-block:: fortran
+
+        subroutine my_subroutine()
+            integer, dimension(10, 10) :: A
+            integer :: i
+            integer :: j
+            $!omp target
+            do i = 1, 10
+                do j = 1, 10
+                    A(i, j) = 0
+                end do
+            end do
+            $!omp end target
+        end subroutine
 
     '''
-    Adds an OpenMP directive to a loop. The
-    optional 'reprod' argument in the apply method decides whether
-    standard OpenMP reduction support is to be used (which is not
-    reproducible) or whether a manual reproducible reproduction is
-    to be used.
+    def apply(self, nodes, options=None):
+        ''' Insert a OMPTargetDirective before the provided node or list
+        of nodes.
+
+        :param nodes: the PSyIR node or nodes to enclose in the OpenMP \
+                      target region.
+        :type nodes: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
```

Comment:
'dict' and 'str' please.

Reply:
Done

I will bring up in next meeting that we probably should use PEP 484 formatting, this should be: 
`Optionl[Dict[str, Any]]`